### DISCUSSION TITLE
Remote message in onNotificationOpenedApp null with data only message?

### DISCUSSION THREAD

**Discussion by oogushikun** [invertase/react-native-firebase#4637] [Q&A]
I'm trying to send a data only message but listeners like onNotificationOpenedApp show the remoteMessage being null.
```
const payload = {
  data: {
    additionalProp1: "string",
    additionalProp2: "string",
    additionalProp3: "string",
    notificationType: "test",
    userId: "01",
    transactionId: "01",
    title: 'test',
    body: 'another test.'
  }
};

const options = {
  priority: "high",
  contentAvailable: true,
  },
  apns: {
    payload: {
      aps: {
        contentAvailable: true
      }
    },
    headers: {
      'apns-push-type': 'background',
      'apns-priority': '5',
    }
  }
}
```

but when I send a notification and data message, remoteMessage isn't null.

```
const payload = {
  notification: {
    title: 'new test',
    body: 'another test.',
  },
  data: {
    additionalProp1: "string",
    additionalProp2: "string",
    additionalProp3: "string",
    notificationType: "test",
    userId: "01",
    transactionId: "01",
    title: 'test',
    body: 'another test.'
  }
};

const options = {
  priority: "high",
  contentAvailable: true,
  },
  apns: {
    payload: {
      aps: {
        contentAvailable: true
      }
    },
    headers: {
      'apns-push-type': 'background',
      'apns-priority': '5',
    }
  }
}
```
Tried going through older issues but can't see a similar issue

**Comment by mikehardy** [comment]
Hi there!
For messaging issues:
- test only on a real device, we've seen cases where the JSON elements were different between emulator and real device
- make sure your versions are up to date v10.1.0 here and ... this is an issue not a discussion but you opened a discussion and thus avoided our issue template so I have no idea what platform or versions. Be up to date.
- test only by posting raw JSON to the REST APIs, and if you have a problem, share the raw JSON you post here, that way we know exactly what you're sending
- if you have a problem, create a reproducible demo with https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh and NOTHING but a single App.js that shows the problem.

**Reply by oogushikun** [reply]
Thanks for the quick reply
Sorry, created an issue with a bit more of some description


But I tested on a physical device and it's still having the same problem.

cant use the latest versions because of outdated dependecies, these are the versions of the packages i've installed
    "@react-native-firebase/analytics": "^7.6.0",
    "@react-native-firebase/app": "^8.4.1",
    "@react-native-firebase/crashlytics": "^8.4.2",
    "@react-native-firebase/messaging": "^7.8.3"

I'm testing by using firebase-admin-server not rest

Let me mak

**Reply by mikehardy** [reply]
I repeat myself:

```

    test only on a real device, we've seen cases where the JSON elements were different between emulator and real device
    make sure your versions are up to date v10.1.0 here and ... this is an issue not a discussion but you opened a discussion and thus avoided our issue template so I have no idea what platform or versions. Be up to date.
    test only by posting raw JSON to the REST APIs, and if you have a problem, share the raw JSON you post here, that way we know exactly what you're sending
    if you have a problem, create a reproducible demo with https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh and NOTHING but a single App.js that shows the problem.

```

Your issue skipped the template almost entirely, I closed it as invalid
